input/include/NovelRT/Input/IInputDevice.hpp

```diff
@@ -29,6 +32,9 @@ namespace NovelRT::Input
         [[nodiscard]] virtual NovelKey& GetAvailableKey(const std::string& keyRequested) = 0;
         [[nodiscard]] virtual NovelRT::Maths::GeoVector2F GetMousePosition() = 0;
         [[nodiscard]] virtual NovelRT::Utilities::Misc::Span<InputAction> GetAllMappings() = 0;
+        [[nodiscard]] virtual KeyStateFrameChangeLog GetCurrentChangeLog(const std::string& key) = 0;
+        [[nodiscard]] virtual KeyStateFrameChangeLog GetPreviousChangeLog(const std::string& key) = 0;
+        [[nodiscard]] virtual std::vector<std::unordered_map<int32_t, KeyStateFrameChangeLog>> GetAllChangeLogs() = 0;
```

Comment:
Should probably be a span of maps. Also, can this getter not be `const` or are we concerned about the fact this is pure virtual?

Reply:
Dunno what the concern would be since they're virtual - i've made them non-const, should be updated on next commit pushed.